src/graft_manager.cpp

```diff
@@ -241,6 +243,22 @@ void BaseTask::onTooBusy()
     respondAndDie("Thread pool overflow");
 }
 
+const char* BaseTask::getStrStatus()
+{
+#define CASE(x) case Status::x: return #x;
```

Comment:
Perhaps a better approach:
```C++
#define GRAFT_STATUS_LIST(EXP) \
    EXP(Ok) \
    EXP(Forward) \
    EXP(InternalError) \
    EXP(Error) \
    EXP(Drop) \
    EXP(Stop)

#define EXP_TO_ENUM(x) x,
#define EXP_TO_STR(x) #x,

enum class Status { GRAFT_STATUS_LIST(EXP_TO_ENUM) };

const char* BaseTask::getStrStatus(Status s)
{
    static const char *status_str[] = { GRAFT_STATUS_LIST(EXP_TO_STR) };
    return status_str[s];
}
```

Reply:
I don't like this. As for me it is difficult to read. And there is no flexibility, for example, we will not be able to set values explicitly, like Forward=5. or use partial set of values.

Reply:
Fixed